RELEASE.md

```diff
@@ -0,0 +1,178 @@
+
+Backwards compatibility is achieved by [bundling multiple task versions](https://docs.microsoft.com/en-us/azure/devops/extend/develop/integrate-build-task?view=azure-devops#bundle-multiple-versions-of-buildrelease-tasks-within-one-extension)
+in the extension and maintaining [task versions](#task-versioning) separately
+from the overall extension version.
+
+### Release versioning
+Every release version number consists of three parts (x.y.z) that
+have a different meaning and describe the type of change this version
+introduces.
+
+- **Patch:** (x.y.**z**): A patch version only includes fixes and small patches.
+- **Minor:** (x.**y**.z): A minor version includes new features that do **NOT**
+  introduce a breaking change.
+- **Major:** (**x**.y.z): A major version is the only version that can contain
+  breaking changes.
+
+Version numbers will be calculated based on commit messages. For that purpose
+please refer to [conventional
+commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) to correctly
+categorize changes for the next version number.
+
+### Pre-release versioning
+Prereleases will add an additional version identifier in the format of
+x.y.z.**p** where the first three version numbers follow the same convention
+as [releases](#release-versioning) and the last version number identifies a pre-release
+among all pre-releases based on the same release.
+
+### Task versioning
+Azure DevOps extension's [tasks are versioned](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/tasks?view=azure-devops&tabs=yaml#task-versions)
+independently from the extension itself.
+Each task has its own manifest file in `task.json` within the task directory
+where the version information of the task is represented using
+[semver](https://semver.org/), for example:
+```
+"version": {
+    "Major": 2,
+    "Minor": 0,
+    "Patch": 0
+  },
+```
+Steps used in Azure DevOps pipeline have a reference to the major version of
+tasks under the assumption that all versions of a task with the same major
+version are backward-compatible and will not be automatically bumped to a newer
+major version.
+
+Task versions **must** be increased when developing features or fixing issues.
+In case of breaking changes to a task we need to duplicate the task files in a
+new directory on disk, increase the major version both in the directory name
+and `task.json` file and then implement the breaking change in the new version.
+
+## How to release
+
+### Typical case: create a new version based on latest release
+
+New versions are created using `Pre-Release` and `Release` workflows available
+in the [`Actions`
+tab](https://github.com/keptn-sandbox/keptn-jenkins-library/actions) on GitHub.
+The worflows will take care of calculating the next version number and create a
+GitHub release depending on the type of changes introduced.
+
+Workflows must be run on the appropriate branch (`master` should be the normal
```

Comment:
Let's state this as `master`/`main` (in case we switch to `main` as the default branch)